src/playlist.js

```diff
@@ -220,6 +210,37 @@ export const sumDurations = function(playlist, startIndex, endIndex) {
 };
 
 /**
+ * Determines the media index of the segment corresponding to the safe edge of the live
+ * window which is the duration of the last segment plus 2 target durations from the end
+ * of the playlist.
+ *
+ * @param {Object} playlist
+ *        a media playlist object;
+ * @return {Number}
+ *         The media index of the segment
```

Comment:
Might need something like "Unless segment length does not exist, in which case returns 0."